### PR TITLE
CASSANDRASC-107: Improve logging for slice restore task

### PR DESCRIPTION
**Pull request by frankgh**
Patch by Francisco Guerrero; Reviewed by TBD for CASSANDRASC-107

### CODE REVIEW

File: CHANGES.txt [hunk 1]
```diff
@@ -82,4 +83,4 @@
  * Add integration tests task (CASSANDRA-15031)
  * Add support for SSL and bindable address (CASSANDRA-15030)
  * Autogenerate API docs for sidecar (CASSANDRA-15028)
- * C* Management process (CASSANDRA-14395)
\ No newline at end of file
+ * C* Management process (CASSANDRA-14395)
```
Comment (yifan-c) [inline CHANGES.txt:86]: what is the change? new line?

File: src/main/java/org/apache/cassandra/sidecar/restore/RestoreSliceTask.java [hunk 2]
```diff
@@ -274,20 +264,16 @@ private CompletableFuture<File> downloadSlice(Promise<RestoreSlice> event)
         }
 
         LOGGER.info("Begin downloading restore slice. sliceKey={}", slice.key());
-        CompletableFuture<File> future = s3Client
-        .downloadObjectIfAbsent(slice)
-        .whenComplete((file, cause) -> {
-            if (cause != null)
+        Future<File> future =
+        fromCompletionStage(s3Client.downloadObjectIfAbsent(slice)).onFailure(cause -> {
```
Comment (yifan-c) [inline src/main/java/org/apache/cassandra/sidecar/restore/RestoreSliceTask.java]: `onFailure` on new line.

File: src/main/java/org/apache/cassandra/sidecar/restore/StorageClient.java [hunk 3]
```diff
@@ -156,6 +159,8 @@ public CompletableFuture<File> downloadObjectIfAbsent(RestoreSlice slice)
         File object = objectPath.toFile();
         if (object.exists())
         {
+            LOGGER.debug("Skipping download, file already exists. jobId={}, s3_object={}",
```
Comment (yifan-c) [inline src/main/java/org/apache/cassandra/sidecar/restore/StorageClient.java]: remove `,`
```suggestion
            LOGGER.debug("Skipping download, file already exists. jobId={} s3_object={}",
```

File: src/main/java/org/apache/cassandra/sidecar/restore/StorageClient.java [hunk 4]
```diff
@@ -257,10 +271,14 @@ CompletableFuture<GetObjectResponse> subscribeRateLimitedWrite(Path destinationP
         }
         catch (FileAlreadyExistsException fileAlreadyExistsException)
         {
+            LOGGER.debug("Skipping download, file already exists. jobId={}, s3_object={}",
```
Comment (yifan-c) [inline src/main/java/org/apache/cassandra/sidecar/restore/StorageClient.java]: remove `,`

**Review by yifan-c — COMMENTED** [review]
some nits + talked offline.

File: CHANGES.txt (hunk 1 above)
Comment (frankgh) [reply CHANGES.txt:86]: yeah, new line, I think this new line was reverted in the prior commit, and my IDE added it back :(

File: src/main/java/org/apache/cassandra/sidecar/restore/RestoreSliceTask.java (hunk 2 above)
Comment (frankgh) [reply src/main/java/org/apache/cassandra/sidecar/restore/RestoreSliceTask.java]: meant to do that! thanks for noticing

**Pull request merged**